server/hotrod/src/main/java/org/infinispan/server/hotrod/BaseRequestProcessor.java

```diff
@@ -0,0 +1,43 @@
+package org.infinispan.server.hotrod;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.Executor;
+
+import io.netty.channel.Channel;
+
+/**
+ * This class is not thread safe. The expected use is to set {@link #cdc}, call some method in inheriting classes
```

Comment:
Link should be fixed?

Reply:
Oh, the whole doc is wrong... At first I didn't realize that the CacheDecodeContext can be accessed by a different thread later on, so I've reused it but I had to revert this. Good catch.